### PR TITLE
Added template for bug report, feature request, and pull request

### PR DESCRIPTION
**Pull request by grgprarup**
## Description
This PR adds the template file for the bug reports, feature requests, and pull requests. 

## Related Issue

- Fixes https://github.com/gherlint/gherlint/issues/22

## Motivation and Context
The template makes it easier for contributors to raise issues and make a pull request to the repo.

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Tests only (no source changes)
- [x] Documentation only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Documentation updated

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+---
+
+## Describe the bug
+
+A clear and concise description of what the bug is.
+
+## Steps to reproduce
+
+Steps to reproduce the behavior:
+
```
Comment (saw-jan) [inline .github/ISSUE_TEMPLATE/bug_report.md:17]: Title only should be fine
```suggestion
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+<details>
+<p>
+
+```console
+version=somevalue
+```
+
+</p>
+</details>
```
Comment (kiranparajuli589) [inline .github/ISSUE_TEMPLATE/bug_report.md:32]:
```suggestion
OS: Ubuntu 20.045
NodeJS Version: v16.0.0
Package Manager: `npm`, `yarn`, etc
Package Manager Version: v1.0.0
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+---
+
+## Is your feature request related to a problem? Please describe.
+
+## Describe the solution you'd like
+
+## Describe alternatives you've considered
+
+## What value does this feature add to the gherlint?
```
Comment (kiranparajuli589) [inline .github/ISSUE_TEMPLATE/feature_request.md:16]:
```suggestion
## What value does this feature add to the GherLint?
```

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+<!--
+Thanks for submitting a change to gherlint!
```
Comment (kiranparajuli589) [inline .github/pull_request_template.md:2]:
```suggestion
Thanks for submitting a change to GherLint!
```

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+To make it possible for us to get your change reviewed and merged please carefully fill out the requested information below.
+
+Please note that any kind of change needs first be submitted to the master branch which holds the next version of gherlint.
+
+Please set the following labels:
+
+- Set label "Status:Needs-Review" for review or "Status:In-Progress" if the PR still has open tasks.
```
Comment (kiranparajuli589) [inline .github/pull_request_template.md:10]: are these status actually available in the repo?

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+- Assignment: assign to self
+- Reviewers: pick at least one
+-->
+
+## Description
+
```
Comment (kiranparajuli589) [inline .github/pull_request_template.md:16]:
```suggestion
<!--- a good and descriptive information about the proposed changes -->
```

Comment (grgprarup) [reply .github/pull_request_template.md:16]: Suggested not to keep these kinds of comments as we have contributing guidelines.
@saw-jan @kiranparajuli589

File: .github/pull_request_template.md (hunk 5 above)
Comment (grgprarup) [reply .github/pull_request_template.md:10]: This goes under the project so I think no need here. I will remove them for now.

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
+---
+name: Feature request
+about: Suggest an idea for improving GherLint
+title: ''
+labels: Category:Feature, Topic:needs-triage
+assignees: ''
+
```
Comment (kiranparajuli589) [inline .github/ISSUE_TEMPLATE/feature_request.md:7]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
+---
+name: Feature request
+about: Suggest an idea for improving GherLint
+title: ''
+labels: Category:Feature, Topic:needs-triage
```
Comment (kiranparajuli589) [inline .github/ISSUE_TEMPLATE/feature_request.md:5]:
```suggestion
tags: enhancement
```
maybe tagging is useful for filtration

**Review by kiranparajuli589 — COMMENTED** [review]
minor nitpicks

**Review by kiranparajuli589 — APPROVED** [review]

**Pull request merged**